core/src/network/event.rs

```diff
@@ -149,7 +149,7 @@ where
 
         /// The handler that was passed to `dial()`, if the
         /// connection failed before the handler was consumed.
-        handler: THandler,
+        handler: Option<THandler>,
```

Comment:
I would remove this field altogether here, as it is confusing and I don't think it is useful in anyway.
It is confusing in `pool.rs` as well, but it is more tolerable there as this is a more internal struct.

Alternatively, if we leave, then we should clarify exactly in which situations this is `Some` and in which this is `None`.

Reply:
Removing it sounds good to me, I never found it particularly useful.